### PR TITLE
Chore: Translate admin helpers to TS

### PR DESCRIPTION
**Pull request by rique223**
Translated the admin helpers.js file to typescript, changed some types on the orchestrator to better use already existing types and changed the heap size of the typecheck script for it to stop overflowing.

### CODE REVIEW

File: packages/core-typings/src/Apps.ts [hunk 1]
```diff
@@ -27,6 +27,27 @@ export type Tiers = {
...
+	seats: number;
+	maxSeats: number;
+	license: License;
+	startDate: string;
+	periodEnd: string;
+	endDate: string;
+	externallyManaged: boolean;
+	isSubscribedViaBundle: boolean;
+};
+
```
Comment (ggazzo) [inline packages/core-typings/src/Apps.ts:50]: what do you think about keep as prefix like AppLicense and AppSubscriptionInfo ?

**Review by ggazzo — CHANGES_REQUESTED** [review]

**Review by matheuslc — APPROVED** [review]
LGTM 🚀

File: packages/core-typings/src/Apps.ts [hunk 2]
```diff
@@ -45,10 +66,11 @@ export type App = {
 	version: string;
 	price: number;
 	purchaseType: string;
-	pricingPlans: PricingPlan[];
+	pricingPlans: AppPricingPlan[];
 	iconFileContent: string;
 	installed?: boolean;
 	isEnterpriseOnly?: boolean;
+	isPurchased?: string;
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:73]: Why is this a string?

File: packages/core-typings/src/Apps.ts [hunk 3]
```diff
@@ -10,21 +10,42 @@ export type AppScreenshot = {
...
-	price: number;
+export type AppLicense = {
+	license: string;
+	version: number;
+	expireDate: string;
+};
+
+export type AppSubscriptionInfo = {
+	typeOf: string;
+	status: string;
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:38]: This is an enum:
```
export enum PurchaseSubscriptionStatus {
    Trialing = 'trialing',
    Active = 'active',
    Cancelled = 'cancelled',
    Cancelling = 'cancelling',
    PastDue = 'pastDue',
}
```

File: packages/core-typings/src/Apps.ts [hunk 4]
```diff
@@ -10,21 +10,42 @@ export type AppScreenshot = {
...
+	statusFromBilling: boolean;
+	isSeatBased: boolean;
+	seats: number;
+	maxSeats: number;
+	license: AppLicense;
+	startDate: string;
+	periodEnd: string;
+	endDate: string;
+	externallyManaged: boolean;
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:47]:
```suggestion
```
This property will not be exposed in the near future.

File: packages/core-typings/src/Apps.ts [hunk 5]
```diff
@@ -45,10 +66,11 @@ export type App = {
 	version: string;
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:66]:
```
detailedDescription: IDetailedDescription;
detailedChangelog: IDetailedChangelog;
```
These are missing, there were defined above and got lost.

File: packages/core-typings/src/Apps.ts [hunk 6]
```diff
@@ -59,11 +81,13 @@ export type App = {
 	marketplaceVersion: string;
 	latest: App;
 	status?: AppStatus;
+	subscriptionInfo: AppSubscriptionInfo;
 	licenseValidation?: {
 		errors: { [key: string]: string };
 		warnings: { [key: string]: string };
 	};
 	marketplace: unknown;
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:89]: What is this property?

File: packages/core-typings/src/Apps.ts [hunk 7]
```diff
@@ -10,21 +10,42 @@ export type AppScreenshot = {
...
+	maxSeats: number;
+	license: AppLicense;
+	startDate: string;
+	periodEnd: string;
+	endDate: string;
+	externallyManaged: boolean;
+	isSubscribedViaBundle: boolean;
 };
 
 export type App = {
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:51]: Is this an AppOverview or AppInfo? AppOverview contains information about the app and the AppInfo contains information about the version of an app. This seems to be a mixture of everything and it honestly worries me.

File: packages/core-typings/src/Apps.ts [hunk 8]
```diff
@@ -45,10 +66,11 @@ export type App = {
...
 	price: number;
 	purchaseType: string;
-	pricingPlans: PricingPlan[];
+	pricingPlans: AppPricingPlan[];
 	iconFileContent: string;
 	installed?: boolean;
 	isEnterpriseOnly?: boolean;
+	isPurchased?: string;
 	isSubscribed: boolean;
 	bundledIn: {
```
Comment (graywolf336) [inline packages/core-typings/src/Apps.ts:75]: Also missing `bundledAppIds: string[]` if this is an AppOverview. If this is an AppInfo, then this property won't exist.

**Review by graywolf336 — DISMISSED** [review]
I have a lot of questions about the `core-typings/src/Apps.ts`. Internally in Marketplace we have AppOverview and AppInfo. While they relate to the same app, they contain vastly different information. One is about the app as a whole (hence `Overview`) and the other is information about a version (hence `Info`). However, this typing is a mixture of both in addition to other information. Can someone explain what is going on?

**Comment by ggazzo** [comment]
> 

well, the types were not checked internally, they were built based on the code/usage.

clearly you are the most knowledgeable person on the subject, what do you think of a fix commit?

File: packages/core-typings/src/Apps.ts (hunk 2 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:73]: Oversight, will be solving this on the next commit.

File: packages/core-typings/src/Apps.ts (hunk 3 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:38]: Solving this on the next commit.

File: packages/core-typings/src/Apps.ts (hunk 4 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:47]: Solving it on the next commit.

File: packages/core-typings/src/Apps.ts (hunk 5 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:66]: detailedDescription was defined on the spot. 
detailedChangelog wasn't being used and, consequently, not defined, I'll be defining it on the next commit for future endeavors.

File: packages/core-typings/src/Apps.ts (hunk 6 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:89]: Not sure, we'll have to discover and correct it on a future PR.

File: packages/core-typings/src/Apps.ts (hunk 7 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:51]: Indeed it looks weird. Let's have a discovery on this and solve it.

File: packages/core-typings/src/Apps.ts (hunk 8 above)
Comment (rique223) [reply packages/core-typings/src/Apps.ts:75]: Also, something to be discovered about.

**Review by ggazzo — CHANGES_REQUESTED** [review]
Please check the translations, I'm afraid you are missing some

**Comment by rique223** [comment]
> Please check the translations, I'm afraid you are missing some

Those are being translated on the "view" side of things, that's why I skipped it, I imagine that if I return a translated string from there it'll bug the translation function since it'll try to translate twice.

**Pull request merged**